pkg/api/node.go

```diff
@@ -121,6 +125,64 @@ func (m *nodeMetrics) Get(ctx context.Context, name string, opts *metav1.GetOpti
 	return &nodeMetrics[0], nil
 }
 
+func (m *nodeMetrics) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1beta1.Table, error) {
+	var table metav1beta1.Table
+
+	switch t := object.(type) {
+	case *metrics.NodeMetrics:
+		table.ResourceVersion = t.ResourceVersion
+		table.SelfLink = t.SelfLink
+		addNodeMetricsToTable(&table, *t)
+	case *metrics.NodeMetricsList:
+		table.ResourceVersion = t.ResourceVersion
+		table.SelfLink = t.SelfLink
+		table.Continue = t.Continue
+		addNodeMetricsToTable(&table, t.Items...)
+	default:
+	}
+
+	return &table, nil
+}
+
+func addNodeMetricsToTable(table *metav1beta1.Table, nodes ...metrics.NodeMetrics) {
+	var names []string
+	for i, node := range nodes {
+		if names == nil {
+			for k := range node.Usage {
+				names = append(names, string(k))
+			}
+			sort.Strings(names)
+
+			table.ColumnDefinitions = []metav1beta1.TableColumnDefinition{
+				{Name: "Name", Type: "string", Format: "name", Description: "Name of the resource"},
+			}
+			for _, name := range names {
+				table.ColumnDefinitions = append(table.ColumnDefinitions, metav1beta1.TableColumnDefinition{
+					Name:   name,
+					Type:   "string",
+					Format: "quantity",
+				})
+			}
+			table.ColumnDefinitions = append(table.ColumnDefinitions, metav1beta1.TableColumnDefinition{
```

Comment:
I would not recommend adding window, as MS returns incorrect value for it. 
MS always return 30s, even though it should differ due scrape frequency and kubelet calculation window `10-15s`.

Reply:
k, I noticed that you want to implement same behaviour in Custom Metrics Adapter. If so it makes more sense.